basiclti/basiclti-common/src/test/org/sakaiproject/basiclti/util/SakaiBLTIUtilTest.java

```diff
@@ -303,14 +295,42 @@ public void testFindBestTool() {
 			bestSite = (String) bestTool.get(LTIService.LTI_SITE_ID);
 			assertEquals(s, bestLaunch);
 			assertEquals(siteId, bestSite);
-
-			bestTool = SakaiBLTIUtil.findBestToolMatch(mostSpecific, tools);
-			bestLaunch = (String) bestTool.get(LTIService.LTI_LAUNCH);
-			bestSite = (String) bestTool.get(LTIService.LTI_SITE_ID);
-			assertEquals(s, bestLaunch);
-			assertEquals(siteId, bestSite);
 		}
+	}
+
+	@Test
+	public void testLabsterUrlMatching() {
+		List<Map<String,Object>> tools = new ArrayList<Map<String,Object>>();
+		Map<String,Object> tool = new HashMap<String,Object>();
+		Map<String,Object> bestTool = null;
+		String bestLaunch, bestSite;
+
+		String siteId = "lti-test";
+		String launchA = "https://api.example.edu/lti/0103940";
+		String launchB = "https://api.example.edu/lti/8493820";
+		String launchC = "https://api.example.edu/lti/5420496";
+		String launchD = "https://api.example.edu/lti/5420496?ignore=me";
+
+		// Lets add the first two to existing tools Map
+		tool.put(LTIService.LTI_LAUNCH, launchA);
+		tool.put(LTIService.LTI_SITE_ID, siteId);
+		tools.add(tool);
+
+		tool.put(LTIService.LTI_LAUNCH, launchB);
+		tool.put(LTIService.LTI_SITE_ID, siteId);
+		tools.add(tool);
+
+		// Lets test match on the third. There should not be a match on an existing URL!
+		bestTool = SakaiBLTIUtil.findBestToolMatch(launchC, tools);
+		Assert.assertNull(bestTool);
```

Comment:
@csev, previously, this would match on launchB ..... because it has the longest common prefix ... that doesn't make sense to me